torchdata/datapipes/iter/transform/bucketbatcher.py

```diff
@@ -10,18 +13,27 @@
 T_co = TypeVar("T_co", covariant=True)
 
 
-def _in_batch_shuffle_fn(data: DataChunk):
-    d = list(data)
-    random.shuffle(d)
-    return DataChunk(d)
+@functional_datapipe("in_batch_shuffle")
+class InBatchShufflerIterDataPipe(IterDataPipe[DataChunk[T_co]]):
+    r"""
+    Shuffles each mini-batches from prior DataPipe (functional name: ``in_batch_shuffle``).
+    """
+
+    def __init__(self, datapipe: IterDataPipe[DataChunk[T_co]]):
+        self.datapipe = datapipe
+
+    def __iter__(self) -> Iterator[DataChunk[T_co]]:
+        for batch in self.datapipe:
+            random.shuffle(batch)
+            yield batch
```

Comment:
Can we add a `__len__` function too? Return from `self.datapipe` when available.